xonsh/environ.py

```diff
@@ -2250,23 +2249,41 @@ def foreign_env_fixes(ctx):
         del ctx["PROMPT"]
 
 
+class _RcPath(str):
```

Comment:
Is str safely subclassable?

Reply:
Yes, it is! I searched online and the oldest reference to this I could find is the 2001 [PEP 253](https://www.python.org/dev/peps/pep-0253/#abstract), which proposed the ability to do this. In the abstract, it notes that this has already been added to Python (though not exactly as proposed there, but that is besides the point). Python has no issues with you extending any of the builtin types, overriding anything you want. In this case, we are just creating a sort of alias, since we are not overriding anything, we just want to be able to identify a specific string.